File: frontend/src/app/models/employee.ts
// export interface Employee{
//     id?:number;
//     name: string;
//     department:string;
//     email: string;
//     sal: number;
// }

export interface EmployeeSearchRequest {
  name?: string;
  email?: string;
  department?: string;
  page: number;
  size: number;
  sortBy: string;
  sortDir: string;
}

export interface Employee {
  id?: number;
  name: string;
  email: string;
  department: string;
  sal: number;
  employmentType?: string;
  joinDate?: string;
}

File: backend/employee-service/src/main/java/com/employee/employeeservice/entity/Employee.java
package com.employee.employeeservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "employees")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String email;

    private String department;
    private Double sal;
    private String profileImageUrl;
    private String status; // ACTIVE, ONBOARDING, OFFBOARDING, TERMINATED

    private java.time.LocalDate joiningDate;
    private String techStackSummary;

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "employee_roles", joinColumns = @JoinColumn(name = "employee_id"))
    @Column(name = "role")
    @Builder.Default
    private java.util.Set<String> roles = new java.util.HashSet<>();
}

